echoregions/regions2d/regions2d.py

```diff
@@ -358,122 +356,129 @@ def plot(
-        region_ids : list
+        region_id : list
             list IDs of regions to create mask for
-        mask_var : str
+        mask_name : str
             If provided, used to name the output mask array, otherwise `mask`
-        mask_labels:
-            None: assigns labels automatically 0,1,2,...
-            "from_ids": uses the region ids
-            list: uses a list of integers as labels
 
         Returns
         -------
-        A DataArray with the data_var masked by the specified region.
+        mask_3d : Data Array
+            A 3D mask where each layer of the mask will contain a 1s/0s mask for each
+            unique label in the 2D mask. The layers will be labeled via region_id values.
+            The slices of the 3d array will be in the form of 1s/0s: masked areas, and
+            non-masked areas.
         """
-        if isinstance(region_ids, list):
-            if len(region_ids) == 0:
-                raise ValueError("region_ids is empty. Cannot be empty.")
+        if isinstance(region_id, list):
+            if len(region_id) == 0:
+                raise ValueError("region_id is empty. Cannot be empty.")
+        elif region_id is None:
+            # Extract all region_id values
+            region_id = self.data.region_id.astype(int).to_list()
         else:
             raise TypeError(
-                f"region_ids must be of type list. Currently is of type {type(region_ids)}"
+                f"region_id must be of type list. Currently is of type {type(region_id)}"
             )
 
-        if isinstance(mask_labels, list) and (len(mask_labels) != len(region_ids)):
-            raise ValueError(
-                "If mask_labels is a list, it should be of same length as region_ids."
-            )
-
-        # Replace nan depth in regions2d.
-        self.replace_nan_depth(inplace=True)
-
         # Dataframe containing region information.
-        region_df = self.select_region(region_ids)
+        region_df = self.select_region(region_id)
 
-        # Select only columns which are important.
+        # Select only important columns
         region_df = region_df[["region_id", "time", "depth"]]
 
-        # Organize the regions in a format for region mask.
-        df = region_df.explode(["time", "depth"])
+        # Filter rows with depth values within self min and self max depth
+        region_df = region_df[
+            (
+                (
+                    region_df["depth"].apply(
+                        lambda x: all(self.min_depth <= i <= self.max_depth for i in x)
+                    )
+                )
+                & (region_df["depth"].apply(lambda x: all(i >= 0 for i in x)))
+            )
+        ]
 
-        # Convert region time to integer timestamp.
-        df["time"] = matplotlib.dates.date2num(df["time"])
+        if region_df.empty:
+            print(
+                "Post NaN Depth Filtered Regions is Empty. All rows had NaN Depth values."
+            )
```

Comment:
Small word tweak to clarify what the filtering operation is.
```suggestion
            print(
                "All rows in has NaN Depth values after filtering depth between min_depth and max_depth."
            )
```